mlflow/data/pandas_dataset.py

```diff
@@ -0,0 +1,143 @@
+                          dataset, including: name, digest, source, and source
+                          type.
+        :return: A string dictionary containing the following fields: name,
+                 digest, source, source type, schema (optional), profile
+                 (optional).
+        """
+        base_dict.update(
+            {
+                "schema": json.dumps({"mlflow_colspec": self.schema.to_dict()}),
+                "profile": json.dumps(self.profile),
+            }
+        )
+        return base_dict
+
+    @property
+    def source(self) -> FileSystemDatasetSource:
+        """
+        TODO: Pandas docs
+        """
+        return self._source
+
+    @property
+    def df(self) -> pd.DataFrame:
+        return self._df
+
+    @property
+    def profile(self) -> Optional[Any]:
+        """
+        TODO: Pandas docs
+        """
+        return {
+            "num_rows": len(self._df),
+            "num_elements": int(self._df.size),
+        }
+
+    @property
+    def schema(self) -> Schema:
+        """
+        An MLflow ColSpec schema representing the columnar dataset
+        """
+        return _infer_schema(self._df)
+
+    def to_pyfunc(self) -> PyFuncInputsOutputs:
+        """
+        Converts the dataset to a collection of pyfunc inputs and outputs for model
+        evaluation. Required for use with mlflow.evaluate().
+        May not be implemented by all datasets.
+        """
+        if self._targets:
+            inputs = self._df.drop(columns=[self._targets])
+            outputs = self._df[self._targets]
+            return PyFuncInputsOutputs(inputs, outputs)
+        else:
+            return PyFuncInputsOutputs(self._df)
+
+
+def from_pandas(
+    df: pd.DataFrame,
+    source: Union[str, DatasetSource],
+    name: Optional[str] = None,
+    digest: Optional[str] = None,
+) -> PandasDataset:
+    """
+    :param df: A Pandas DataFrame
+    :param source: The source from which the DataFrame was derived, e.g. a filesystem
+                    path, an S3 URI, an HTTPS URL, a delta table name with version, or
+                    spark table etc. Attempting to use other source types will throw.
+    :param targets: An optional target column name or list of target column names for
+                    supervised training. The columns must be present in the dataframe
+                    (`df`).
+    :param name: The name of the dataset. If unspecified, a name is generated.
+    :param digest: A dataset digest (hash). If unspecified, a digest is computed
+                    automatically.
+    """
+    from mlflow.data.dataset_source_registry import resolve_dataset_source
+
+    if isinstance(source, DatasetSource):
+        resolved_source = source
+    else:
+        resolved_source = resolve_dataset_source(
+            source,
+            candidate_sources=[FileSystemDatasetSource, DeltaDatasetSource, SparkDatasetSource],
```

Comment:
Can we get away with not specifying candidate sources here? Any source should work